Sources/SPMTestSupport/PackageGraphTester.swift

```diff
@@ -230,13 +234,19 @@ public final class ResolvedProductResult {
     }
 
     public func checkDerivedPlatforms(_ platforms: [String: String], file: StaticString = #file, line: UInt = #line) {
-        let targetPlatforms = Dictionary(uniqueKeysWithValues: product.platforms.derived.map({ ($0.platform.name, $0.version.versionString) }))
+        // To retain existing behavior, we have to check both the product type, as well as the types of all of its targets.
```

Comment:
is this comment correct everywhere? there is a few place in this PR that check only for `product.type == .test`

Reply:
That's a good question actually. I arrived at this conclusion fairly late, will go back and verify all the other call sites as well.

Reply:
There's no impact on any of the scenarios we're testing, but it seems more correct to actually always do this since that is essentially how the previous code worked. I'm going to move this to a `linksXCTest` computed property on resolved products so that all call sites can do the right thing easily.